### DISCUSSION TITLE
Sound selection fails or hangup on some devices

### DISCUSSION THREAD

**Discussion by cybernissart** [Navid200/xDrip#87] [Alerts]
Observed that system sound selection operation fails and revert back to previous screen instantaneously and sometimes hang with roughly 1mn delay, instead of giving access to the file  system. Preventing access to Predictive features.

GUI > Settings > Alerts and Alarms > Extra Alerts > 'Persistent High' or 'Predictive low' sound settings
GUI > Settings > Alerts and Alarms > Other Alerts > 'Alert Sound ' sound settings

It can be reproduced in latest nightly build, but last for month for following device:
Xiaomi Redmi note 9/10 pro running Android 11

It works correctly on Samsung S5 running Android 7.
It also works correctly for following method and presents file system drawer
GUI > Settings > Alerts and Alarms > Glucose Level Alerts list' sound settings

Any review of the file selector method used that explain this difference?

**Comment by cybernissart** [comment]
Video showing contact attached

**Reply by Navid200** [reply]
I don't see a video.
I think there is a limit in the size of a clip you can upload.  If you could upload to youtube, it would be best.
Or, you can put the clip up on facebook, and I will see if I can upload it to youtube if you don't know how to do it.

**Reply by cybernissart** [reply]
Correct, trying to find a way to share the screen record

**Comment by cybernissart** [comment]
https://user-images.githubusercontent.com/24457611/139367306-b60a740e-3877-46b4-8bb5-b71013666b75.mp4

**Reply by Navid200** [reply]
Great.  One question that will come up is if you are using an official version of xDrip and if yes, which version.

**Reply by cybernissart** [reply]
Latest xDrip+ nightly build c98a3c4

**Comment by cybernissart** [comment]
Working with Glucose level menu.

https://user-images.githubusercontent.com/24457611/139367448-068609d1-644d-440f-a577-43844c7afb2f.mp4

**Comment by Navid200** [comment]
@jamorham @tolot27 What could cause choosing to change the sound for persistent high alert to just go back to the main screen?
I have never seen this happen.  It doesn't happen on my device.  But, I don't have Xiaomi.
Have you guys seen anything like this before?

**Reply by cybernissart** [reply]
I also observed it and reported on a thread channel somewhere, as I was using a Sony Xperia years ago, but then I broke it , and went to another mobile type that did not show the issue anymore.
Issue is long lasting.
Difference in the code between the two file selection functions.

**Comment by Navid200** [comment]
I have reopened the issue.  Please let's continue over there.
https://github.com/NightscoutFoundation/xDrip/issues/1862
